controllers/secure.go

```diff
@@ -107,13 +107,18 @@ func (c *SecureContext) submitRequest(rw http.ResponseWriter, req *http.Request,
 	}
 
 	// Get RemoteAddr from the request
-	clientIP, err := GetClientIP(req)
-	if err == nil {
-		// Set headers for requests to CF API proxy
-		request.Header.Add("X-Client-IP", clientIP)
-		request.Header.Add("X-TIC-Secret", c.Settings.TICSecret)
-	} else {
-		log.Println(err)
+	if c.Settings.TICSecret != "" {
+		clientIP, err := GetClientIP(req)
+		if err != nil {
```

Comment:
`if err != nil {` -> `if err != nil || clientIP == ""`

Reply:
But `GetClientIP` can return an empty string and a `nil` error. This happens when the proxied request has no `RemoteAddr`, when we generate a request like this: https://github.com/18F/cg-dashboard/blob/master/controllers/uaa.go#L280-L282.